core/lib/multivm/src/versions/vm_fast/tests/rollbacks.rs

```diff
@@ -142,3 +145,33 @@ fn test_vm_loadnext_rollbacks() {
 
     assert_eq!(result_without_rollbacks, result_with_rollbacks);
 }
+
+#[test]
+fn rollback_in_call_mode() {
+    let counter_bytecode = read_test_contract();
+    let counter_address = Address::repeat_byte(1);
+
+    let mut vm = VmTesterBuilder::new()
+        .with_empty_in_memory_storage()
+        .with_execution_mode(TxExecutionMode::EthCall)
+        .with_custom_contracts(vec![ContractToDeploy::new(
+            counter_bytecode,
+            counter_address,
+        )])
+        .with_random_rich_accounts(1)
+        .build();
+    let account = &mut vm.rich_accounts[0];
+    let tx = account.get_test_contract_transaction(counter_address, true, None, false, TxType::L2);
+
+    let (compression_result, vm_result) = vm
+        .vm
+        .execute_transaction_with_bytecode_compression(tx, true);
+    compression_result.unwrap();
+    assert_matches!(
+        vm_result.result,
+        ExecutionResult::Revert { output }
+            if output.to_string().contains("This method always reverts")
+    );
+    // The new VM performs deduplication internally and doesn't record reads, hence this crude comparison.
```

Comment:
Just removed a comment to avoid confusion. You can see a non-crude version in the `vm_latest` counterpart; it checks that there are no deduplicated write logs.